### ISSUE TITLE
How do I evaluate MACD factor?

### ISSUE THREAD

**Issue by DanielKui** [microsoft/qlib#2058]
>> from qlib.data.dataset.loader import QlibDataLoader
>> MACD_EXP = '2 * ((EMA($close, 12) - EMA($close, 26))/$close - EMA((EMA($close, 12) - EMA($close, 26))/$close, 9))'
>> fields = [MACD_EXP] # MACD
>> names = ['MACD']
>> labels = ['Ref($close, -2)/Ref($close, -1) - 1'] # label
>> label_names = ['LABEL']
>> data_loader_config = {
..     "feature": (fields, names),
..     "label": (labels, label_names)
.. }
>> data_loader = QlibDataLoader(config=data_loader_config)
>> df = data_loader.load(instruments='csi300', start_time='2010-01-01', end_time='2017-12-31')
>> print(df)

I leanred from https://qlib.readthedocs.io/en/latest/advanced/alpha.html , saw MACD factor, How do I calc its IC?
How should I create DataHandlerLP and dataset based on MACD? anyone give me an example? Should I have to create class inherited from Alpha158, then store it in handler.py? is there any simplest way?
Thank you very much

**Comment by SunsetWolf** [comment]
Hi, @DanielKui 

You don’t need to inherit from `Alpha158` to compute IC for a custom factor like MACD.
`Qlib` already allows building a handler directly from an expression using `DataHandlerLP + QlibDataLoader`.

Here is a sample based on official data:

```
from qlib.data.dataset.handler import DataHandlerLP
from qlib.data.dataset import DatasetH


data_handler_config = {
    "start_time": "2008-01-01",
    "end_time": "2020-08-01",
    "instruments": "csi300",
    "data_loader": {
        "class": "QlibDataLoader",  # Assuming QlibDataLoader is a string reference to the class
        "kwargs": {
            "config": {
                "feature": [["2 * ((EMA($close, 12) - EMA($close, 26))/$close - EMA((EMA($close, 12) - EMA($close, 26))/$close, 9))"], ["MACD"]],
                "label": [['Ref($close, -2)/Ref($close, -1) - 1'], ["LABEL"]],
            },
            "freq": "day",
        },
    },
}

segments = {
    "train": ["2008-01-01", "2014-12-31"],
    "valid": ["2015-01-01", "2016-12-31"],
    "test": ["2017-01-01", "2020-08-01"],
}


data_handler = DataHandlerLP(**data_handler_config)
ds = DatasetH(handler=data_handler, segments=segments)

df = ds.prepare("test")
print(df)

ic_series = df.groupby("datetime").apply(lambda x: x["MACD"].corr(x["LABEL"]))
print(f"Mean IC = {ic_series.mean()}")
print(f"IC series: \n\n{ic_series}")

```

This directly loads the MACD factor and computes its IC — no need to create a custom handler class or modify handler.py.
